activerecord/lib/active_record/store.rb

```diff
@@ -136,6 +166,36 @@ def store_accessor_for(store_attribute)
         type_for_attribute(store_attribute.to_s).accessor
       end
 
+      class AccessorType # :nodoc
+        def self.type_for_accessor(type_name)
+          case type_name
+          when Class
+            type_name.new
+          when Symbol
+            klass = standard_types.fetch(type_name, Type::Value)
+            klass.new
+          else
+            type_name
+          end
+        end
+
+        private
+
+        def self.standard_types
```

Comment:
Also, perhaps deprecate `store_accessor` method name in favor of `store_attribute`? It'd be more in-line with new `attribute` API, and also make more semantic sense because accessor implies a purely in-memory ruby getter/setter, while `attribute` macro does casting and interacts with db. 

Alternatively this could be `attribute :something, store: true` or `attribute :something, store: { some_attr: :some_type }`.

CC @sgrif